go/test/endtoend/vtgate/queries/informationschema/informationschema_test.go

```diff
@@ -31,127 +31,135 @@ import (
 	"vitess.io/vitess/go/test/endtoend/cluster"
 )
 
+func start(t *testing.T) (utils.MySQLCompare, func()) {
+	mcmp, err := utils.NewMySQLCompare(t, vtParams, mysqlParams)
+	require.NoError(t, err)
+
+	deleteAll := func() {
+		_, _ = utils.ExecAllowError(t, mcmp.VtConn, "set workload = oltp")
+
+		tables := []string{
+			"t1", "t1_id2_idx", "vstream_test", "t2", "t2_id4_idx", "t3", "t3_id7_idx", "t4",
+			"t4_id2_idx", "t5_null_vindex", "t6", "t6_id2_idx", "t7_xxhash", "t7_xxhash_idx", "t7_fk", "t8",
+		}
+		for _, table := range tables {
+			_, _ = mcmp.ExecAndIgnore("delete from " + table)
+		}
+	}
+
+	deleteAll()
+
+	return mcmp, func() {
+		deleteAll()
+		mcmp.Close()
+		cluster.PanicHandler(t)
+	}
+}
+
 func TestDbNameOverride(t *testing.T) {
-	defer cluster.PanicHandler(t)
-	ctx := context.Background()
-	conn, err := mysql.Connect(ctx, &vtParams)
-	require.Nil(t, err)
-	defer conn.Close()
-	qr, err := conn.ExecuteFetch("SELECT distinct database() FROM information_schema.tables WHERE table_schema = database()", 1000, true)
+	mcmp, closer := start(t)
+	defer closer()
+
+	qr, err := mcmp.VtConn.ExecuteFetch("SELECT distinct database() FROM information_schema.tables WHERE table_schema = database()", 1000, true)
 
 	require.Nil(t, err)
 	assert.Equal(t, 1, len(qr.Rows), "did not get enough rows back")
 	assert.Equal(t, "vt_ks", qr.Rows[0][0].ToString())
 }
 
 func TestInformationSchemaQuery(t *testing.T) {
-	defer cluster.PanicHandler(t)
-	ctx := context.Background()
-	conn, err := mysql.Connect(ctx, &vtParams)
-	require.NoError(t, err)
-	defer conn.Close()
-
-	utils.AssertSingleRowIsReturned(t, conn, "table_schema = 'ks'", "vt_ks")
-	utils.AssertSingleRowIsReturned(t, conn, "table_schema = 'vt_ks'", "vt_ks")
-	utils.AssertResultIsEmpty(t, conn, "table_schema = 'NONE'")
-	utils.AssertSingleRowIsReturned(t, conn, "table_schema = 'performance_schema'", "performance_schema")
-	utils.AssertResultIsEmpty(t, conn, "table_schema = 'PERFORMANCE_SCHEMA'")
-	utils.AssertSingleRowIsReturned(t, conn, "table_schema = 'performance_schema' and table_name = 'users'", "performance_schema")
-	utils.AssertResultIsEmpty(t, conn, "table_schema = 'performance_schema' and table_name = 'foo'")
-	utils.AssertSingleRowIsReturned(t, conn, "table_schema = 'vt_ks' and table_name = 't1'", "vt_ks")
-	utils.AssertSingleRowIsReturned(t, conn, "table_schema = 'ks' and table_name = 't1'", "vt_ks")
+	mcmp, closer := start(t)
+	defer closer()
+
+	utils.AssertSingleRowIsReturned(t, mcmp.VtConn, "table_schema = 'ks'", "vt_ks")
```

Comment:
nit: it would be nice with a comment saying why we are not checking these assertions against mysql